risc0/circuit/bigint/src/byte_poly.rs

```diff
@@ -96,6 +96,17 @@ impl BytePoly {
         Self::from_biguint(rem, coeffs)
     }
 
+    pub fn nondet_inv<Rhs: Borrow<BytePoly>>(&self, rhs: Rhs, coeffs: usize) -> BytePoly {
+        // Computes the inverse of LHS mod RHS via the `inv = lhs^{rhs - 2} % rhs` algorithm
+        // TODO: Is num_bigint's `modpow` algorithm efficient?
```

Comment:
Looks reasonably efficient, it uses 4-bit windowed exponentiation using Montgomery reduction for odd RHS, otherwise for even RHS it uses plain binary modular exponentiation. Modular inverse using extended Euclidean algorithm would be much more efficient on average as numbers tend to get small fast. One concern is (maybe, for some reason) the algorithm must be constant-time, in this case Euclidean algorithm wouldn't work and neither does the num_bigint's modpow which is also non-constant time.

Reply:
Thank you! This all sounds good to me